### ISSUE TITLE
Get AWS guidance on Sender ID for international SMS

### ISSUE THREAD

**Issue by yaelberger-commits** [cds-snc/notification-planning#1065]
_Get AWS guidance on Sender ID for international SMS_

## Description

Reach to AWS for support and open an ADR to document information their experts share with us about sending SMS internationally.

### WHY are we building?

Singapore has a Jan. 31, 2023 deadline for registering a Sender ID or SMS marked as "LIKELY-SCAM"
After 6 months, SMS will be blocked

### WHAT are we building?

- [x] ADR

### VALUE created by our solution

We'll have a better idea of what's involved with registering a Sender ID to send to international countries like Singapore, India and Philippines 

> AWS warning
> You are receiving this notice because you have sent at least one SMS message to a Singapore phone number in the last 90 days. If you are currently not sending SMS to Singapore and/or do not plan to send SMS further, you do not need to take any further action.
> 
> If you are currently sending or plan to start sending SMS messages to Singapore phone numbers, you must start using a Sender ID and register it with both AWS and Singapore SMS Sender ID Registry (SSIR) prior to January 31, 2023. A Sender ID is an alphanumeric originator that identifies the sender of an SMS message, e.g. “AMAZON”. If you do not take this action prior to January 31, 2023, your SMS messages to Singapore will be impacted. You are required to take this action to comply with new regulations announced by Singapore regulator Infocomm Media Development Authority (IMDA). Starting January 31, 2023 we will start replacing all unregistered Sender IDs and generic ones such as “NOTICE” or “Secure” with “Likely-SCAM” when sending SMS to Singapore, as per IMDA requirement. Six months from January 31, 2023, we will start blocking all SMS traffic to Singapore with generic or unregistered Sender IDs, as per IMDA requirement. If you use Sender IDs already registered with Singapore SSIR and AWS, you do not need to take any further action.
> 
> We want to emphasize that you will need to register your Singapore Sender ID(s) with both AWS and SSIR. We will make a Sender ID registration form available in the AWS Console in January 2023 to enable you to complete your AWS portion of the registration. We will send additional announcements on the date of availability of this registration form.
> 
> At this time we ask that you obtain a Singapore unique entity number (“UEN”) if you don’t already have one. You will need this UEN to register your Sender ID(s) with both AWS and SSIR. You can review if your company has a UEN by using the following public search [1]. If you are a business located outside Singapore, you can still obtain a UEN by registering with Singapore’s Accounting and Corporate Regulatory Authority (“ACRA”). Depending on your situation, getting a UEN with ACRA can take a few weeks, so we strongly recommend that you start now. While you need to register your Sender ID(s) with SSIR as well, we ask that you do NOT register with SSIR until you register with AWS first. Registering with SSIR without a corresponding registration with AWS in place, can lead to your SMS messages getting blocked.
> 
> If you require further assistance, reach out to AWS Support [2].
> [1] https://www.uen.gov.sg/ueninternet/faces/pages/uenSrch.jspx
> [2] https://aws.amazon.com/support

## Acceptance Criteria** (Definition of done) 

- [x] Given we have clients who wish to send SMS to Singapore or another country that requires a registered Sender ID, when We need to register a Sender ID, then the team knows the steps involved and how to do it.
- [x] Start on an ADR that documents requirements different country's constraints on sender IDs and how to handle these on our level.

## QA Steps

- [x] Team feedback on ADR

**Comment by jimleroyer** [comment]
Just opened an AWS ticket to as a few questions on this:
https://support.console.aws.amazon.com/support/home#/case/?displayId=11801337331&language=en

**Comment by yaelberger-commits** [comment]
Jimmy will review AWS response

**Comment by jimleroyer** [comment]
The response from AWS support for the Singapore case is interesting. It seems they suggest we could just have a long code for Singapore and use it to send messages to Singapore, to bypass the sender ID registration. I will reply to clarify on that.

> I understand that since you are not a business but operating as the government of Canada you are wondering if there is a shorter way to accomplish the Sender ID registration for Singapore and also to verify the understanding on the steps to get a Sender ID.
> 
> As per the steps that you have shared in the previous correspondence, the understanding is correct. The 3 Steps that needs to be followed are -
> 
> Step 1. Registering for a Singapore Unique Entity Number (UEN)
> Step 2. If you are requesting the sender ID in the Amazon Pinpoint regions, use the Amazon Pinpoint Sender ID registration instructions to register the sender ID. To complete the registration process manually, use the Singapore sender ID registration instructions.
> Step 3. Registering a Sender ID with Singapore Network Information Centre (SGNIC)
> 
> To get UEN, Local companies and businesses in Singapore can apply through ACRA. However, for other entities outside Singapore, you can get a UEN by applying through different government agencies such as:
> 
> International Enterprise Singapore
> Ministry of Finance
> Ministry of Education
> Ministry of defence
> Ministry of foreign affairs
> 
> Islamic Religious Council of Singapore
> Ministry of Health
> Ministry of Communication and Information
> Ministry of Law
> Monetary Authority
> 
> Ministry of Culture, Community, and Youth
> People’s Association
> Registry of Societies
> Ministry of Manpower
> Singapore Land Authority
> 
> Smart Nation and Digital Government Office
> Ministry of National Development.
> 
> I hope this approach simplifies things for alternate options to obtain a UEN number.
> 
> You can still procure a Long Code or Short Code to send out SMS messages to Singapore instead of a Sender ID if you do not wish to go through the registration process. To do so, please raise a new case with the Limit increase team for dedicated Long Code for Singapore, based on the availability of Long Codes will be assigned and further requirements if any will be asked by the Limit increase team. Steps for raising this case are provided in the link (1) mentioned below in the reference section.
> 
> Please feel free to reach out to me if you have any further queries and I will be more than happy to help. Take care and enjoy the rest of your day!
> 
> Reference Links:
> [1]. Requesting dedicated long codes for SMS messaging with Amazon Pinpoint - Requesting a long code - https://docs.aws.amazon.com/pinpoint/latest/userguide/channels-sms-awssupport-long-code.html#request-number-row

**Comment by andrewleith** [comment]
- @jimleroyer to reply to AWS today

**Comment by jimleroyer** [comment]
I replied back to AWS, asking if we can simply use a long code from Singapore and skip sender ID registration, as last reply seemed to imply. I want to make sure I understood correctly.

**Comment by jimleroyer** [comment]
AWS support got back to me and confirmed that we can simply request a Singapore long code in order to bypass SENDER IDs. Most probably this will get identified as AWS as a Sender ID. I will reply back and ask. Here is the full reply response:

>> This is helpful but might not make it easier for us. If there would be one organization with a simpler process, which one would that be?

> Since this is new process that has been rolled out recently Singapore Telecommunication governing body, AWS is not aware of any such entity with a simpler process in among the list that was shared, if we do find any such information we will definitely make it available for you and in public documentation to help all our customers. Since this process is always carried out by our customers at their end, we do not have any information or similar cases to refer to for this info.

>> Oh wait!? We can simply get long codes from Singapore to send to Singapore numbers? without the need for registering a sender ID? If yes, that would incredibly simplify the process and might be easier for us to exclusively send from these numbers instead. Can you confirm if I understood you correctly?

> Yes, as mentioned in my previous response, you can raise a request with the Limit Increase team for Long Code for Singapore [1]. Once the request is placed based on the current requirement, the Limit increase team will verify and should be able to assign a Long Code number to your account in the requested region. Once a Long Code is procured and assigned to your account you should be able to use this number to send out SMS messages and you would not require to go through the Sender ID process, unless your business use case demands strictly a Sender ID.

**Comment by andrewleith** [comment]
- Waiting on reply from AWS Support

**Comment by yaelberger-commits** [comment]
Jimmy will check response this morning
Looks like we can just order a long code from AWS for Singapore, but would have a sender ID Amazon

**Comment by smcmurtry** [comment]
- @jimleroyer had a call with AWS support, resulting in some confusion
- support seems confident that the long code solution will work
- @yaelberger-commits suggests we wrap this up and pull back to the product backlog
- @jimleroyer will write ADR for this and wrap up

**Comment by yaelberger-commits** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/1065?planning-poker) with Zenhub @jimleroyer

**Comment by jimleroyer** [comment]
Written ADR along with recommendation for Singapore:
https://github.com/cds-snc/notification-adr/pull/34/files

tl;dr: Don't register sender ID, we're not using them so no need. Monitor situation to see if our long code notifications still can reach out to the destination.

**Comment by smcmurtry** [comment]
Yael reviewed, everything looked good, left a comment in github regarding the date.
Back to Jimmy for a last look.

**Comment by jimleroyer** [comment]
I addressed Yael's comment and added her recommended change.

**Comment by yaelberger-commits** [comment]
Moving to done
